secret-provider/src/lib.rs

```diff
@@ -3,10 +3,53 @@ pub mod file;
 pub mod k8s;
 
 use error::Error;
+use tracing::trace;
 
 use async_trait::async_trait;
+use platform::{current_platform_type, PlatformType};
 use serde::de::DeserializeOwned;
 
+/// Parse and return encryption parameters from a Kubernetes secret or a file.
+pub async fn params_get_from_secret_source<T: DeserializeOwned + Send>(
+    secret_source_name: &str,
+) -> Result<T, Error> {
+    let secret_params = match current_platform_type() {
+        PlatformType::K8s => {
+            let client = kube::Client::try_default()
+                .await
+                .map_err(|e| Error::KubeApi { source: e })?;
+            // todo: Add cli arg for namespace as well?
+            let ns = client.default_namespace().to_string();
```

Comment:
If running from cli I guess we could read something like env `MY_POD_NAMESPACE` ?

```suggestion
            let ns = match std::env::var("MY_POD_NAMESPACE") {
              Ok(ns) if !ns.is_empty() => ns,
              _ => client.default_namespace().to_string()
            };
```

Reply:
> If running from cli I guess we could read something like env `MY_POD_NAMESPACE` ?

Will that be relevant here though if we are running out of k8s cluster? we'll not take this route.
Or you're saying from a perspective if we want to for example edit daemonset and pass this env?

Reply:
MY_POD_NAMESPACE looks like a k8s standard env, so might conflict with the correct (`client.default_namespace()`) namespace as well?

Reply:
io-engine can be run outside of a k8s pod as long as the connections are setup correctly
Up till now, io-engine had no k8s connection, so this namespace was not required, but in this case we have to specify correct namespace if we want it to find the secrets.
If you have a look at platform, there's already uses of `MY_POD_NAMESPACE`.

Reply:
done